winton_kafka_streams/state/state_store.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar, Generic
+
+from ..processor.serialization import Serde
+from .key_value_state_store import KeyValueStateStore
+
+KT = TypeVar('KT')  # Key type.
+VT = TypeVar('VT')  # Value type.
+
+
+class StateStore(ABC, Generic[KT, VT]):
+    """
+    StateStores are created by Suppliers for use in StreamTasks
+    """
+    def __init__(self, name: str, key_serde: Serde[KT], value_serde: Serde[VT], logging_enabled: bool) -> None:
+        self.logging_enabled: bool = logging_enabled
+        self._value_serde: Serde[VT] = value_serde
+        self._key_serde: Serde[KT] = key_serde
+        self._name: str = name
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def serialize_key(self, key: KT) -> bytes:
+        return None
+        return self._key_serde.serializer.serialize("", key)
```

Comment:
Two returns?

Reply:
Oops. Good catch! Fixed.
This has been corrected in my next branch already and has a test in place.